cocotb/utils.py

```diff
@@ -196,7 +202,13 @@ def unpack(ctypes_obj, string, bytes=None):
         :exc:`ValueError`: If length of *string* and size of *ctypes_obj*
             are not equal.
         :exc:`MemoryError`: If *bytes* is longer than size of *ctypes_obj*.
+
+    .. deprecated:: 1.5
+        This function is deprecated. The functionality is a duplicate of :py:class:`struct.Struct`.
```

Comment:
I don't agree with this comment. I think we're still right to deprecate it, but `ctypes.Structure` and `struct.Struct` are fairly incompatible APIs.

I think an exact replacement is:
```
memoryview(ctypes_obj).cast('B')[:bytes] = string
```
while something the user might also be looking for one of
* `ctypes.pointer(ctypes_obj).contents = type(ctypes_obj).from_buffer_copy(string)`
* `obj = MyStruct.from_buffer_copy(string)`

Reply:
```suggestion
        This function is deprecated. Use use ``memoryview(ctypes_obj)[:bytes] = string`` instead.
        You may also be looking for the :func:`~ctypes._CData.from_buffer_copy` method of :mod:`ctypes` types,
        if you are trying to convert bytes to a structure, or ``ctypes.pointer(obj).contents = new_obj`` if you are trying
        to modify a structure in place.
```